### PR TITLE
Fix DSC Community Plaster template

### PR DESCRIPTION
**Pull request by johlju**
# Pull Request

This updates the Plaster template module type 'dsccommunity' to use the latest pipeline files. The correct azure-pipelines.yml and Codecov.io support. It also makes a build.yml cleaner when using that module type. 

### Added
- Added a new template file for azure-pipelines.yml when using the
  module type `'dsccommunity'`.
- Added a new template and configuration for Codecov.io when using
  module type `'dsccommuntity'`.

### Changed
- Updated the template GitVersion.yml to use specific words to bump
  major version (previously it bumped if the word was found anywhere in
  the commit message even if it was part of for example a code variable).
- Updated the template file build.yaml to make it more clean when using
  the module type `'dsccommunity'`.

### Fixed
- Fixed a whitespace issue in the template file Resolve-Dependency.psd1.
- Rephrased comments in the template file build.yaml.

## Task list

- [x] The PR represents a single logical change. i.e. Cosmetic updates should go in different PRs.
- [ ] Added an entry under the Unreleased section of in the CHANGELOG.md as per [format](https://keepachangelog.com/en/1.0.0/).
- [ ] Local clean build passes without issue or fail tests (`build.ps1 -ResolveDependency`).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [ ] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

---
This change is [Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/206)

### CODE REVIEW

File: Sampler/Templates/Build/build.yaml.template [hunk 1]
```diff
@@ -66,11 +84,25 @@ NestedModule:
 ####################################################
 #       Sampler Pipeline Configuration             #
 ####################################################
+<%
+    if ($PLASTER_PARAM_ModuleType -eq 'dsccommunity') {
+@"
+BuildWorkflow:
+  '.':
+    - build
+    - test
```
Comment (gaelcolas) [inline Sampler/Templates/Build/build.yaml.template:93]: It looks like when the `ModuleType -eq 'dsccommunity'` we're not doing `hqrmtest` anymore, am I right? is that Intentional?

File: Sampler/Templates/Sampler/CHANGELOG.md.template [hunk 2]
```diff
@@ -1,26 +1,30 @@
-# Changelog
+# Changelog for {ModuleName}
```
Comment (gaelcolas) [inline Sampler/Templates/Sampler/CHANGELOG.md.template:1]: did you mean `# Changelog for <%=$PLASTER_PARAM_ModuleName%>`?

File: Sampler/Templates/Sampler/CHANGELOG.md.template [hunk 3]
```diff
@@ -1,26 +1,30 @@
...
-All notable changes to this project will be documented in this file.
-
-The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+The format is based on and uses the types of changes according to [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
 ### Added
+
```
Comment (gaelcolas) [inline Sampler/Templates/Sampler/CHANGELOG.md.template:9]: I don't think this is
- 1. valid keepachangelog
- 2. valid md

I think titles need a newline after, but I may be wrong.

File: Sampler/Templates/Sampler/plasterManifest.xml [hunk 4]
```diff
@@ -491,7 +488,7 @@
    <!-- CHANGELOG MD -->
    <file source='CHANGELOG.md.template'
```
Comment (gaelcolas) [inline Sampler/Templates/Sampler/plasterManifest.xml:489]: With the comment above using `$PLASTER_PARAM_ModuleName` we should change the type from `<file ` to `<templateFile `.

**Review by gaelcolas — CHANGES_REQUESTED** [review]
I think the few comments need double checking, but nothing major.

**Comment by gaelcolas** [comment]
FTR, test failing because missing in dsccommunity template integration test:
```
        	.github
        	CHANGELOG.md
        	.github/ISSUE_TEMPLATE
        	.github/PULL_REQUEST_TEMPLATE.md
        	.github/ISSUE_TEMPLATE/General.md
        	.github/ISSUE_TEMPLATE/Problem_with_resource.md
        	.github/ISSUE_TEMPLATE/Resource_proposal.md

```

File: Sampler/Templates/Build/build.yaml.template (hunk 1 above)
Comment (johlju) [reply Sampler/Templates/Build/build.yaml.template:93]: It is doing hqrmtest in a job in the azure-pipelines, but not default when running task 'test'. Need to run `-Task hqrmtest` to run them from console. So it is intentionally as is this way in all the modules we converted.

File: Sampler/Templates/Sampler/CHANGELOG.md.template (hunk 2 above)
Comment (johlju) [reply Sampler/Templates/Sampler/CHANGELOG.md.template:1]: I just fixed that in anther file (github template), so can fix it here too. Can do in another PR.

File: Sampler/Templates/Sampler/CHANGELOG.md.template (hunk 3 above)
Comment (johlju) [reply Sampler/Templates/Sampler/CHANGELOG.md.template:9]: It should be a valid keepachangelog, and the sections should be okay.
Real world one here: https://raw.githubusercontent.com/dsccommunity/SqlServerDsc/master/CHANGELOG.md

File: Sampler/Templates/Sampler/CHANGELOG.md.template (hunk 2 above)
Comment (gaelcolas) [reply Sampler/Templates/Sampler/CHANGELOG.md.template:1]: If I don't beat you to it ;)

File: Sampler/Templates/Sampler/plasterManifest.xml (hunk 4 above)
Comment (johlju) [reply Sampler/Templates/Sampler/plasterManifest.xml:489]: Done

File: Sampler/Templates/Sampler/CHANGELOG.md.template (hunk 2 above)
Comment (johlju) [reply Sampler/Templates/Sampler/CHANGELOG.md.template:1]: Done

**Comment by johlju** [comment]
Hopefully it will pass now. Run it locally and it updates change log as expected (with the correct module name).

**Review by gaelcolas — APPROVED** [review]

**Pull request merged**